agent/acs/client/acs_client_test.go

```diff
@@ -361,7 +238,49 @@ func TestConnectClientError(t *testing.T) {
 
 	cs := New(testServer.URL, testCfg, credentials.AnonymousCredentials)
 	err := cs.Connect()
-	if _, ok := err.(*wsclient.WSError); !ok || err.Error() != "InvalidClusterException: Invalid cluster" {
-		t.Error("Did not get correctly typed error: " + err.Error())
-	}
+	_, ok := err.(*wsclient.WSError)
+	assert.True(t, ok)
+	assert.EqualError(t, err, "InvalidClusterException: Invalid cluster")
+}
+
+func testCS(conn *mock_wsclient.MockWebsocketConn) wsclient.ClientServer {
+	testCreds := credentials.AnonymousCredentials
+	cs := New("localhost:443", testCfg, testCreds).(*clientServer)
+	cs.SetConnection(conn)
+	return cs
+}
+
+func startMockAcsServer(t *testing.T, closeWS <-chan bool) (*httptest.Server, chan<- string, <-chan string, <-chan error, error) {
```

Comment:
IIRC, it has subtle differences that made it annoying to merge them. I'll take another look, though.